test/mainnet-fork/CCTPCalls.t.sol

```diff
@@ -165,59 +181,88 @@ contract BaseChainUSDCToCCTPTestBase is ForkTestBase {
     function setUp() public override virtual {
         super.setUp();
 
-        destination = getChain("base").createSelectFork(20187000);  // September 24, 2024
+        /*** Step 1: Set up environment and deploy mocks ***/
+
+        destination = getChain("base").createSelectFork(20782500);  // October 7, 2024
 
         usdsBase  = IERC20(address(new ERC20Mock()));
         susdsBase = IERC20(address(new ERC20Mock()));
         usdcBase  = IERC20(USDC_BASE);
 
-        rateProvider = new MockRateProvider();
-
-        rateProvider.__setConversionRate(1.25e27);
+        /*** Step 2: Deploy and configure PSM with a pocket ***/
 
         deal(address(usdsBase), address(this), 1e18);  // For seeding PSM during deployment
 
         psmBase = IPSM3(PSM3Deploy.deploy(
-            admin, USDC_BASE, address(usdsBase), address(susdsBase), address(rateProvider)
+            SPARK_EXECUTOR, USDC_BASE, address(usdsBase), address(susdsBase), SSR_ORACLE
         ));
 
-        foreignAlmProxy   = new ALMProxy(admin);
-        foreignRateLimits = new RateLimits(admin);
+        vm.prank(SPARK_EXECUTOR);
+        psmBase.setPocket(pocket);
+
+        vm.prank(pocket);
+        usdcBase.approve(address(psmBase), type(uint256).max);
+
+        /*** Step 3: Deploy and configure ALM system ***/
 
-        foreignController = new ForeignController({
-            admin_      : admin,
-            proxy_      : address(foreignAlmProxy),
-            rateLimits_ : address(foreignRateLimits),
-            psm_        : address(psmBase),
-            usdc_       : USDC_BASE,
-            cctp_       : CCTP_MESSENGER_BASE
+        ControllerInstance memory controllerInst = ForeignControllerDeploy.deployFull({
+            admin : SPARK_EXECUTOR,
+            psm   : address(psmBase),
+            usdc  : USDC_BASE,
+            cctp  : CCTP_MESSENGER_BASE
         });
 
-        // NOTE: FREEZER, RELAYER, and CONTROLLER are taken from super.setUp()
+        foreignAlmProxy   = ALMProxy(payable(controllerInst.almProxy));
+        foreignRateLimits = RateLimits(controllerInst.rateLimits);
+        foreignController = ForeignController(controllerInst.controller);
+
+        ForeignControllerInit.AddressParams memory addresses = ForeignControllerInit.AddressParams({
+            admin         : SPARK_EXECUTOR,
+            freezer       : freezer,
+            relayer       : relayer,
+            oldController : address(0),  // Empty
+            psm           : address(psmBase),
+            cctpMessenger : CCTP_MESSENGER_BASE,
+            usdc          : USDC_BASE
+        });
 
-        vm.startPrank(admin);
+        RateLimitData memory usdcDepositData = RateLimitData({
```

Comment:
Same here these can be collapsed into 1 or 2 rate limit data structs.

Reply:
Updated